pilot/pkg/networking/core/route/route.go

```diff
@@ -1156,13 +1156,18 @@ func translateHeaderMatch(name string, in *networking.StringMatch) *route.Header
 }
 
 // TranslateCORSPolicy translates CORS policy
-func TranslateCORSPolicy(in *networking.CorsPolicy) *cors.CorsPolicy {
+func TranslateCORSPolicy(proxy *model.Proxy, in *networking.CorsPolicy) *cors.CorsPolicy {
 	if in == nil {
 		return nil
 	}
 
 	// CORS filter is enabled by default
 	out := cors.CorsPolicy{}
+	// Start from Envoy 1.30(istio 1.22), cors filter will not forward preflight requests to upstream by default.
+	if proxy.IstioVersion.Compare(&model.IstioVersion{Major: 1, Minor: 22, Patch: -1}) >= 0 {
```

Comment:
Move this to util similar to how we do for other versions of proxy checks so that this code can be reused and removed easily later